### ISSUE TITLE
misc: support read-only default env-file, with optional user override

### ISSUE THREAD

**Issue by lucab** [Irqbalance/irqbalance#118]
This is a follwup to https://github.com/coreos/fedora-coreos-config/pull/158 and https://bugzilla.redhat.com/show_bug.cgi?id=1752520.

Looking at Fedora packaging of `irqbalance`, I noticed that systemd service unit has been patched to use the default env-file which is placed under `/etc/sysconfig/irqbalance`.

While every downstream distribution is free to arrange configs and paths as they prefer, it would be nice to encourage a better split between read-only distribution defaults and user's customizations.

Purely as an example, this could be achieved by re-arranging the service unit like this:
```
EnvironmentFile=/usr/lib/irqbalance/defaults.env
EnvironmentFile=-/etc/sysconfig/irqbalance
```
Or anything else to the same effect. My personal wishes here are 1) having the distro defaults in a read-only place 2) supporting a possibly empty `/etc` directory.

/cc @nhorman

**Comment by nhorman** [comment]
I'm not comfortable with a requirement for a second environment variable, as doing so would cause breakage to existing installations, but Im fine with adding a second environment file for you.  Would this work for you:
EnvironmentFile=-/usr/lib/irqbalance/defaults.env
EnvironmentFile=-/etc/sysconfig/irqbalance

let me know and I'll check it in

**Comment by lucab** [comment]
I'm sorry as I'm not currently grasping the breakage you are hinting at, but yes I think that setup would work as well. One concern would be that I don't know what happens if both files are missing (i.e. the variables are not set at all).

For clarity, the overall suggestion is for downstream packages to only ship `/usr/lib/irqbalance/defaults.env`, and let the users be free to create and manage `/etc/sysconfig/irqbalance` if they need to customize anything.

**Comment by nhorman** [comment]
The breakage that I'm hinting at is that if I in the upstream project mandate that a new config file be added in /usr/lib, any distribution taking that update then needs to modify their environment to include that file, and in already installed instances, that file won't be there.  Its certainly fixable, but I don't want to place the requirement on downstream consumers, since they already have a config file in /etc.  And so, if I make the change, I want to make the /usr/lib/ config file optional.

I'll check the fix in shortly

**Comment by nhorman** [comment]
fixed in commit 724243f1744a263b3777e920526c0a4a3c042ca6
